src/typesystem.hpp

```diff
@@ -55,6 +55,15 @@ struct PolyData {
 	std::vector<VarId> vars;
 };
 
+// RENAME: 'Kind' isn't the exactly the right terminology here
+enum class kind_type { TypeFunction, Mono, Poly };
+
+// Variable that can contain any type, and even type functions
+struct TypeVarData {
+	kind_type kind;
+	TypeVarId type_var_id;
+};
```

Comment:
Huh, that's interesting. What's it for?

Reply:
Oh, actually, I think I see what this is for. good foresight!

Reply:
It's for `x := <type>`, I don't know if it can be used in some other way

Reply:
yeah, it will probably be useful to be able to handle any type-like thing with a uniform interface, without having to bloat CompileTimeEnvironment with different methods to handle monotypes, polytypes, and type funcs